common/flatpak-dir.c

```diff
@@ -11794,15 +11832,45 @@ flatpak_dir_list_remote_config_keys (FlatpakDir *self,
   return NULL;
 }
 
-char **
+static void
+add_subdirs (GPtrArray *res,
+             GFile     *parent)
+{
+  g_autoptr(GFileEnumerator) dir_enum = NULL;
+
+  dir_enum = g_file_enumerate_children (parent,
+                                        G_FILE_ATTRIBUTE_STANDARD_NAME ","
+                                        G_FILE_ATTRIBUTE_STANDARD_TYPE,
+                                        G_FILE_QUERY_INFO_NONE,
+                                        NULL, NULL);
+  if (dir_enum == NULL)
+    return;
+
+  while (TRUE)
+    {
+      GFileInfo *info;
+      GFile *path;
+
+      if (!g_file_enumerator_iterate (dir_enum, &info, &path, NULL, NULL) ||
+          info == NULL)
+        break;
+
+      if (g_file_info_get_file_type (info) == G_FILE_TYPE_DIRECTORY)
+        g_ptr_array_add (res, g_object_ref (path));
+    }
+}
+
+GPtrArray *
 flatpak_dir_get_sideload_repo_paths (FlatpakDir *self)
 {
-  GKeyFile *config = flatpak_dir_get_repo_config (self);
+  g_autoptr(GFile) sideload_repos_dir = flatpak_dir_get_sideload_repos_dir (self);
+  g_autoptr(GFile) sideload_runtime_repos_dir = flatpak_dir_get_runtime_sideload_repos_dir (self);
+  g_autoptr(GPtrArray) res = g_ptr_array_new_with_free_func (g_object_unref);
 
-  if (config)
-    return g_key_file_get_string_list (config, "core", "xa.sideload-repos", NULL, NULL);
+  add_subdirs (res, sideload_repos_dir);
```

Comment:
I guess it's the job of whoever adds to them to create `/run/flatpak/sideload-repos` and `/var/lib/flatpak/sideload-repos` if they don't exist? I guess then they will be owned by root. That seems alright for the use case Endless has, but it is a difference between this and the old implementation, which allowed non-root users to install into the system installation from a USB if the polkit rules allowed it. So maybe we will eventually want to add a way for the system helper to add to these directories.

Reply:
You can do it with the --sideload-repo=PATH argument. Maybe we should we allow additional configuration in $HOME. I'm not sure how important that is though...

Reply:
Good point, the CLI option seems good enough